src/Service.Tests/Unittests/MultiSourceQueryExecutionUnitTests.cs

```diff
@@ -152,5 +156,64 @@ public async Task TestMultiSourceQuery()
             Assert.AreEqual("db1", queryMap1[0].Value, $"Data returned for {queryName1} is incorrect for multi-source query");
             Assert.AreEqual("db2", queryMap2[0].Value, $"Data returned for {queryName2} is incorrect for multi-source query");
         }
+
+        /// <summary>
+        /// Test to ensure that the correct access token is being set when multiple data sources are used.
+        /// </summary>
+        [TestMethod]
+        public async Task TestMultiSourceTokenSet()
+        {
+            string defaultSourceConnectionString = "Server =<>;Database=<>;";
+            string childConnectionString = "Server =child;Database=child;";
+
+            string dataSourceName1 = "db1";
+            string db1AccessToken = "AccessToken1";
+            string dataSourceName2 = "db2";
+            string db2AccessToken = "AccessToken2";
+
+            Dictionary<string, DataSource> dataSourceNameToDataSource = new()
+            {
+                { dataSourceName1, new(DatabaseType.MSSQL, defaultSourceConnectionString, new())},
+                { dataSourceName2, new(DatabaseType.MSSQL, childConnectionString, new()) }
+            };
+
+            RuntimeConfig mockConfig = new(
+               Schema: "",
+               DataSource: new(DatabaseType.MSSQL, defaultSourceConnectionString, new()),
```

Comment:
What is `new()` being used for? Can you use the param name for better readability?

Reply:
fixed. thanks for review!